runners/spark/pom.xml

```diff
@@ -134,6 +152,28 @@
             <forkCount>1</forkCount>
             <reuseForks>false</reuseForks>
           </configuration>
+          <executions>
+            <execution>
+              <id>runnable-on-service-tests</id>
+                 <configuration>
+                   <groups>org.apache.beam.sdk.testing.RunnableOnService</groups>
+                <parallel>none</parallel>
+                <failIfNoTests>true</failIfNoTests>
+                <dependenciesToScan>
+                  <dependency>org.apache.beam:java-sdk-all</dependency>
+                </dependenciesToScan>
+                <systemPropertyVariables>
+                  <beamTestPipelineOptions>
+                    [
+                      "--runner=org.apache.beam.runners.spark.TestSparkPipelineRunner",
+                      "--streaming=false"
+                    ]
+                  </beamTestPipelineOptions>
+                  <dataflow.spark.test.reuseSparkContext>true</dataflow.spark.test.reuseSparkContext>
```

Comment:
Yea, we definitely don't really want to run all of them all the time, especially if they are too expensive.

In particular, there are three conflated things that should be teased apart somehow:

 - A test that *can* run on a service. Basically, a test that does not depend on in-memory hackery.
 - A test that *should* run on each runner (either in precommit or postcommit) to confirm the capability matrix and that we don't break the runner by messing with the SDK.
 - A test that *requires* a runner but is really an SDK unit test. These are tests, for example, that the composite transforms are correctly assembled. This category is sort of new from moving all runners out of the SDK, and is mostly not that interesting.

The current `RunnableOnService` tag is the first one, but actually that one is the least interesting now IMO. I thought I might get lucky and be able to gain some testing signal prior to properly separating these and organizing the tests by the capabilities they require.